addons/sales_team/tests/test_default_team.py

```diff
@@ -0,0 +1,40 @@
+from odoo.tests import common
+
+
+class TestDefaultTeam(common.SavepointCase):
+    """Tests to check if correct default team is found."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Set up data for default team tests."""
```

Comment:
Not an official policy, but can you make your test class independent from demo data?

Reply:
can you be a bit more specific here?
I should test with new user (not demo user that is provided already)? Or root user (though, it would test ignoring access rights, which would not test real case scenario (if for example, access rights are set up incorrectly)?

P.S. I thought, reusing demo data was a good idea, because it would speed up tests a bit (don't need to recreate, clean up data if needed demo data is already there).

Reply:
It indeed speeds up testing a bit, but you don't have a lot of data to generate and I dislike using demo data for testing purposes (it makes it impossible to use your test in an normal db and makes the test fragile - if another module change demo data in xml files, your test might fail even though they're correct). 

Having little data + a `SavepointCase` is usually enough for perf imho
Indeed, create a user and test with it, create teams, etc. - using _non-demo_ data (e.g. `sales_team.team_sales_department`) is fine, though I don't mind if you create both teams in the test 

see an example here: https://github.com/odoo/odoo/blob/12.0/addons/project/tests/test_project_base.py

Reply:
Created new user, though for teams, it would be quite problematic to workaround, because `sales_team.team_sales_department` is hard coded in `_get_default_team_id`. And removing such team, would make method behave differently.

Reply:
i agree that you need to keep using the fallback data team `sales_team.team_sales_department` in the test to be coherent with the code, however the `sales_team.crm_team_1` should be removed and replaced by another one; sthg like
```python
cls.team_1 = cls.env['crm.team'].create({
'name': 'Test Team',
'member_ids': [(4, cls.user.id)],
'company_id': False
})
```

either you're fully independent from demo data or not, but meeting me halfway here is kinda useless :yum:

Reply:
Well, technically both teams are not coming from demo data, though are updated by demo data:)

Reply:
nope, `team_sales_department` is indeed modified in demo but `crm_team_1` is a demo-only data (which is why I'm insisting here)

Reply:
thx for the change :)

Reply:
Oh right, my mistake. Confused it with website team.